internal/cli/search/index_opts.go

```diff
@@ -170,14 +165,59 @@ func (opts *IndexOpts) NewSearchIndex() (*atlasv2.SearchIndexCreateRequest, erro
 	return i, nil
 }
 
+func (opts *IndexOpts) readCreateFromFile() (*atlasv2.SearchIndexCreateRequest, error) {
```

Comment:
This PR says fixing e2e tests, but there is a functionality change. Is this expected?

Reply:
this is not strictly needed, but I added the file validation so that we could catch this in the unit tests, since loading the index file was just filling the known fields and not failing

Reply:
thanks for clarifying